### PR TITLE
JP-3232: Fix NRC coron duplicate product names

### LINKED ISSUE #7730: Level 3 NIRCam Coronagraphic products have conflicting names
_Issue [JP-3232](https://jira.stsci.edu/browse/JP-3232) was created on JIRA by [Jonathan Eisenhamer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=eisenham):_

With the resolution of JP-3219 through [PR#&#x2060;7556](https://github.com/spacetelescope/jwst/pull/7556), there is now a name conflict with the Level 3 products.

The conflict arises because there are two primary Level 3 products being produced, one through `calwebb_coron3` and one through `calwebb_image3`. The current naming scheme does not have provision to identify which type of level 3 processing has produced products.

Some method, either in the stem part of the Level 3 name, or through some manipulation of suffixes, will need be developed in order to distinguish the two products.

### PR DESCRIPTION
**Pull request by hbushouse**
Resolves [JP-3232](https://jira.stsci.edu/browse/JP-3232)

Closes #7730 

This PR adds an override for the construction of the level-3 product name for NIRCam coronagraphic associations that are to be processed as regular imaging, i.e. through the image3 pipeline. The data are also processed as a coron3 ASN, which previously resulted in identical "i2d" level-3 product names being created: one from coron3 and one from image3. This update breaks the file name degeneracy by simply adding the additional suffix "-image3" to the optical element field of the product name for the data processed as regular imaging.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: jwst/associations/lib/rules_level3.py [hunk 1]
```diff
@@ -474,6 +474,10 @@ def __init__(self, *args, **kwargs):
         # Check and continue initialization.
         super(Asn_Lv3NRCCoronImage, self).__init__(*args, **kwargs)
 
+    @property
```
Comment (hbushouse) [inline jwst/associations/lib/rules_level3.py:477]: For reviewers, this just adds an override for the regular "dms_product_name" function, to be used by only this particular ASN class.

**Comment by hbushouse** [comment]
Regtest run started at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/868

**Review by stscieisenhamer — APPROVED** [review]

**Review by kmacdonald-stsci — APPROVED** [review]

**Comment by stscieisenhamer** [comment]
If there are pools in the artifactory area, the tests themselves should see the change and report failure, along the lines of different/extra/missing products. If the changes are the expected ones, okifying will rectify the changes.

**Comment by hbushouse** [comment]
> If there are pools in the artifactory area, the tests themselves should see the change and report failure, along the lines of different/extra/missing products. If the changes are the expected ones, okifying will rectify the changes.

The jw00016 is in the associations/sdp/pools/ directory on artifactory, which I believe is the input area for all the tests based on sdp-generated pools, but I also see a directory /truth/sdp/pools/ that contains a lot of pool files. 1) Why would there be pool files in a truth area, and 2) if they're necessary, do I need to put a copy of the jw00016 pool in there too?

**Comment by stscieisenhamer** [comment]
That is unfortunate. The tests should be fully self-contained under `assocations`. The semi-duplicate under `truth/associations` should not be there. I'll double verify/repair artifactory.

For your purposes, it is only the top level `associations` folder that is of concern. Note that if the pool already exists, the okify script will handle fixing, so there should be no manual intervention on your part. If you wish to add new pools, they go into `associations/sdp/pools`. For new pools, run the regtests, then okify.

**Comment by hbushouse** [comment]
Regtests are clean.

**Pull request merged**